fix(Select): Update selected on option change

bjosttveit opened the pull request:
When setting an initial value to a select, it currently does not work if the options are loaded after the initial value is, since it will already have run the useEffect without finding the option and is not run again once the options are loaded. In app-fronten-react, this led to select-boxes always showing up empty on refresh even if they had been previously filled out, because option-loading happens later than formData-loading in app-frontend-react.

mimarz commented on packages/react/src/components/Select/Select.tsx:
```diff
@@ -163,7 +163,7 @@ const Select = (props: SelectProps) => {
       setSelectedValues(value ?? []);
     }
-  }, [value]);
+  }, [value, options]);
```
I haven't looked to closely at the code in this component, but is this the correct way to solve this problem? `options` is not used in this effect, so this feels more like a workaround than a fix...? 🤔 

At a quick glance it looks like maybe the effect hook on [L110](https://github.com/digdir/designsystem/blob/cd1f43ccc78b77baf11a4f01221572ffd13470fe/packages/react/src/components/Select/Select.tsx#L110) needs to be updated with a options deps array.

bjosttveit replied:
`options` is used in the sense that `findOptionFromValue` depends on `options`. 

There are many dependencies that are not included in the list, which causes it not to update when it should. There is actually a linting warning asking the list to be changed to:
`[findOptionFromValue, multiple, selectedValues, value]`

Which might also fix the problem.

bjosttveit replied:
I changed it so that it uses all of the hook dependencies now, so no more linting warning

TomasEng replied:
Just to explain the logic behind the hook @mimarz is referring to: Having no dependencies makes it run on every render. However, there is an `if` statement at the start checking whether the following functions should run again. So basically, instead of comparing values in the dependency array, it just runs a normal `if` check, so it shouldn't have a significant impact on the performance. I understand that this is not a very clean way of doing it though, so I'll be happy if you have any suggestions on how to improve it.

bjosttveit commented:
I am trying something else hold on

bjosttveit commented:
Right, I did not realize there where multiple places that updated the state like that, so i took your suggestion and moved the valueChanging logic to the same place as the option-changing logic and deleted the old useEffect.

I have tested in app-frontend and it works as expected there.

bjosttveit commented:
It passes all tests in app-frontend, so I am quite confident now.

![image](https://user-images.githubusercontent.com/47412359/229142446-0acbc399-f949-4cc5-8486-2736bc977dce.png)
![image](https://user-images.githubusercontent.com/47412359/229142651-eb41e97d-4e08-4c6e-9e3b-d82915ed42fc.png)

Anyways, no rush! Happy Easter 🐣

TomasEng reviewed (approved):
Looks good, but maybe you should add a test for this specific use case to ensure that the error doesn't happen again when someone modifies the component?

bjosttveit commented:
> Looks good, but maybe you should add a test for this specific use case to ensure that the error doesn't happen again when someone modifies the component?

Good call, I found a bug when writing the test. It appears to be in conflict with this:
![image](https://user-images.githubusercontent.com/47412359/230017441-90231060-225c-4a32-9d22-b8a34426baf1.png)

If options are loaded late this will change the value to the first option
